### PR TITLE
Feature: support photo selection inside scanning view

### PR DESCRIPTION
**Pull request by henryleunghk**
This PR includes the following change:
- added support of photo selection inside scanning view

Preview:
IMG_3CE46695D536-1

### CODE REVIEW

File: WeScan/ImageScannerController.swift [hunk 1]
```diff
@@ -55,12 +55,21 @@ public final class ImageScannerController: UINavigationController {
         view.translatesAutoresizingMaskIntoConstraints = false
         return view
     }()
+
```
Comment (wetransferplatform) [inline WeScan/ImageScannerController.swift:58]: - :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)

File: WeScan/ImageScannerController.swift [hunk 2]
```diff
@@ -71,6 +80,9 @@ public final class ImageScannerController: UINavigationController {
             navigationBar.tintColor = .black
         }
         navigationBar.isTranslucent = false
+        if (canSelect == true) {
```
Comment (wetransferplatform) [inline WeScan/ImageScannerController.swift:83]: - :warning: `if`, `for`, `guard`, `switch`, `while`, and `catch` statements shouldn't unnecessarily wrap their conditionals or arguments in parentheses. (`control_statement`)

File: WeScan/ImageScannerController.swift [hunk 3]
```diff
@@ -136,7 +148,32 @@ public final class ImageScannerController: UINavigationController {
...
+                    selectPhotoButton.widthAnchor.constraint(equalToConstant: 44.0),
+                    selectPhotoButton.heightAnchor.constraint(equalToConstant: 44.0),
+                    selectPhotoButton.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -24.0),
+                    view.bottomAnchor.constraint(equalTo: selectPhotoButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
+                ]
+            }
+            
+            constraints = constraints + selectPhotoButtonConstraints;
+        }
+
```
Comment (wetransferplatform) [inline WeScan/ImageScannerController.swift:175]: - :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)

**Comment by AvdLee** [comment]
@henryleunghk Thanks for taking the time to contribute to our library. However, I'm afraid this is out of the scope of WeScan. Adding image picker options can be done in many ways and comes with tasks like managing permissions.

We see this as something to add by implementors themselves.

Feel free to open a PR to make the library more flexible to do this, allowing you to add the image picker button by subclassing one of our classes, for example.

File: WeScan/ImageScannerController.swift [hunk 4]
```diff
@@ -40,7 +40,7 @@ public protocol ImageScannerControllerDelegate: NSObjectProtocol {
 /// 1. Uses the camera to capture an image with a rectangle that has been detected.
 /// 2. Edit the detected rectangle.
 /// 3. Review the cropped down version of the rectangle.
-public final class ImageScannerController: UINavigationController {
+public final class ImageScannerController: UINavigationController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
```
Comment (wetransferplatform) [inline WeScan/ImageScannerController.swift:44]: - :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)

File: WeScan/ImageScannerController.swift [hunk 5]
```diff
@@ -136,7 +148,32 @@ public final class ImageScannerController: UINavigationController {
...
+                ]
+            } else {
+                selectPhotoButtonConstraints = [
+                    selectPhotoButton.widthAnchor.constraint(equalToConstant: 44.0),
+                    selectPhotoButton.heightAnchor.constraint(equalToConstant: 44.0),
+                    selectPhotoButton.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -24.0),
+                    view.bottomAnchor.constraint(equalTo: selectPhotoButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
+                ]
+            }
+
```
Comment (wetransferplatform) [inline WeScan/ImageScannerController.swift:172]: - :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)

File: WeScan/ImageScannerController.swift [hunk 6]
```diff
@@ -136,7 +148,32 @@ public final class ImageScannerController: UINavigationController {
...
+                    selectPhotoButton.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -24.0),
+                    view.bottomAnchor.constraint(equalTo: selectPhotoButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
+                ]
+            }
+            
+            constraints = constraints + selectPhotoButtonConstraints;
+        }
+        
+        NSLayoutConstraint.activate(constraints)
     }
```
Comment (wetransferplatform) [inline WeScan/ImageScannerController.swift:178]: - :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)

**Comment by henryleunghk** [comment]
> @henryleunghk Thanks for taking the time to contribute to our library. However, I'm afraid this is out of the scope of WeScan. Adding image picker options can be done in many ways and comes with tasks like managing permissions.
> 
> We see this as something to add by implementors themselves.
> 
> Feel free to open a PR to make the library more flexible to do this, allowing you to add the image picker button by subclassing one of our classes, for example.

Thanks for your feedback!

Do you mean to create a subclass of `ImageScannerController` rather than adding that option inside `ImageScannerController` directly?

**Pull request closed without merging**